packages/fxa-auth-server/fxa-oauth-server/lib/grant.js

```diff
@@ -132,9 +138,10 @@ module.exports.validateRequestedGrant = async function validateRequestedGrant(
 // the specified grant has been sufficiently vetted by calling code.
 module.exports.generateTokens = async function generateTokens(grant) {
   // We always generate an access_token.
-  const access = await db.generateAccessToken(grant);
+  const access = await exports.generateAccessToken(grant);
+
   const result = {
-    access_token: access.token.toString('hex'),
+    access_token: access.jwt_token || access.token.toString('hex'),
```

Comment:
I did that initially, the problem is below we have to de-hex encode it to a buffer so that we can create the hash for looking up the token in the DB. This way ended up being a bunch cleaner.

Reply:
> In fact in the case of issuing a JWT access token, it should be the hash of the whole JWT, not just the jti.

That was another thing I wanted to bring up with the spec author, whether to hash the whole token or whether the at_hash should just be the jti. The whole thing makes more sense to me.